### PR TITLE
daemon-rpc: update output of all calls and their description

### PR DESCRIPTION
**Pull request by erciccione**
I tested all the current calls and updated their description in our RPC guide for the daemon.

~~This is still a work in progress, i tested all the calls but many definitions need to be updated. Any help would be highly appreciated.~~

~~grep `TODO` to find the definitions that still need to be updated.~~

part of #1434

### CODE REVIEW

File: resources/developer-guides/daemon-rpc.md [hunk 1]
```diff
@@ -108,6 +107,7 @@ Outputs:
 
 * *count* - unsigned int; Number of blocks in longest chain seen by the node.
 * *status* - string; General RPC error code. "OK" means everything looks good.
+* *untrusted* - boolean; States if the result is obtained using the bootstrap mode, and is therefore not trusted (`true`), or when the daemon is fully synced (`false`).
```
Comment (moneromooo-monero) [inline resources/developer-guides/daemon-rpc.md:110]: The second part is confusing. Took me a bit to realize it was correct :) Maybe: "or when the daemon is fully synced and thus handles the RPC locally" ?

File: resources/developer-guides/daemon-rpc.md [hunk 2]
```diff
@@ -240,19 +250,29 @@ Outputs:
 
 * *block_header* - A structure containing block header information.
   * *block_size* - unsigned int; The block size in bytes.
+  * *block_weight* - TODO
+  * *cumulative_difficulty* - unsigned int; Cumulative difficulty of all blocks in the blockchain.
```
Comment (moneromooo-monero) [inline resources/developer-guides/daemon-rpc.md:254]: Only up to the block in the reply

File: resources/developer-guides/daemon-rpc.md [hunk 3]
```diff
@@ -842,35 +983,7 @@ $ curl http://127.0.0.1:18081/json_rpc -d '{"jsonrpc":"2.0","id":"0","method":"g
...
     }],
     "status": "OK"
-  }
-}
-```
-
-
-### **flush_txpool**
-
-Flush tx ids from transaction pool
```
Comment (moneromooo-monero) [inline resources/developer-guides/daemon-rpc.md:852]: Why is this one removed ?

**Comment by moneromooo-monero** [comment]
The vast majority of addition are just "TODO", wrong patch uploaded ?

**Comment by erciccione** [comment]
> The vast majority of addition are just "TODO", wrong patch uploaded ?

Sorry i wasn't clear. Most of the TODOs are new responses (like difficulty_top64, next_seed_hash, seed_hash, seed_height, etc) that don't have a definition yet. I left them as TODOs hoping to get the correct definitions from CLI devs.

Thanks for the other suggestions, i will implement them with the next push.

File: resources/developer-guides/daemon-rpc.md (hunk 1 above)
Comment (erciccione) [reply resources/developer-guides/daemon-rpc.md:110]: That's copied from the definition of *unstrusted* in another call. I can change it everywhere, no problem.

File: resources/developer-guides/daemon-rpc.md (hunk 2 above)
Comment (erciccione) [reply resources/developer-guides/daemon-rpc.md:254]: Changing to `unsigned int; Cumulative difficulty of all blocks up to the block in the reply.`

File: resources/developer-guides/daemon-rpc.md (hunk 3 above)
Comment (erciccione) [reply resources/developer-guides/daemon-rpc.md:852]: It gave me an error when trying to call it, but i guess that could be intended. Reverting

**Comment by erciccione** [comment]
So, as i mentioned in my precedent comment, some new definitions were added and need to be described. This is the list of the responses that still need a description (grep `TODO` to find them in this PR):

difficulty_top64
next_seed_hash
seed_hash
seed_height
wide_difficulty
wide_cumulative_difficulty
wide_fee_amount
wide_emission_amount
block_weight
block_weight_limit
block_weight_median
cumulative_difficulty_top64
long_term_weight
credits
long_term_weight
miner_tx_hash
pow_hash
database_size
nettype
update_available
version (get_info)
emission_amount_top64
fee_amount_top64
release
main_chain_parent_block
next_needed_pruning_seed
overview
prunable_as_hex
prunable_hash
pruned_as_hex
bg_idle_threshold
bg_ignore_battery
bg_min_idle_seconds
bg_target
block_reward
block_target
pow_algorithm
fee_total

**Comment by erciccione** [comment]
More definitions added. Thanks @ndorf.

**Comment by erciccione** [comment]
Some missing calls are being added to this pr with https://github.com/erciccione/monero-site/pull/26. We need that PR to be checked by somebody familiar with the daemon before merging it.

**Comment by erciccione** [comment]
This is ready to be reviewed.

File: resources/developer-guides/daemon-rpc.md [hunk 4]
```diff
@@ -239,21 +250,31 @@ Inputs: *None*.
 Outputs:
 
 * *block_header* - A structure containing block header information.
-  * *block_size* - unsigned int; The block size in bytes.
+  * *block_size* - unsigned int; Backward compatibility, same as *block_weight*, use that instead
+  * *block_weight* - unsigned int; The adjusted block size, in bytes. This is the raw size, plus a positive adjustment for any Bulletproof transactions with more than 2 outputs.
+  * *cumulative_difficulty* - unsigned int; Cumulative difficulty of all blocks up to the block in the reply.
```
Comment (moneromooo-monero) [inline resources/developer-guides/daemon-rpc.md:255]: Might want to say "least 64" as you did above for difficulty ?

File: resources/developer-guides/daemon-rpc.md [hunk 5]
```diff
@@ -239,21 +250,31 @@ Inputs: *None*.
...
   * *hash* - string; The hash of this block.
   * *height* - unsigned int; The number of blocks preceding this block on the blockchain.
+  * *long_term_weight* - unsigned int; The long term block weight, based on the median weight of the preceding 100000 blocks.
   * *major_version* - unsigned int; The major version of the monero protocol at this block height.
+  * *miner_tx_hash* - string; The hash of this block's coinbase transaction.
   * *minor_version* - unsigned int; The minor version of the monero protocol at this block height.
   * *nonce* - unsigned int; a cryptographic random one-time number used in mining a Monero block.
   * *num_txes* - unsigned int; Number of transactions in the block, not counting the coinbase tx.
   * *orphan_status* - boolean; Usually `false`. If `true`, this block is not part of the longest chain.
+  * *pow_hash* - string; The hash, as a hexadecimal string, included in the block as proof-of-work.
```
Comment (moneromooo-monero) [inline resources/developer-guides/daemon-rpc.md:268]: Technically, it's not included in the block, but calculated from the block. OK to leave as is though.

File: resources/developer-guides/daemon-rpc.md [hunk 6]
```diff
@@ -1096,7 +1271,10 @@ Inputs: *None*.
 
 Outputs:
 
+* *credits* - unsigned int; If payment for RPC is enabled, the number of credits available to the requesting client. Otherwise, 0.
 * *height* - unsigned int;
+* *next_needed_pruning_seed* - unsigned int; The next pruning seed needed for pruned sync.
+* *overview* - string; Overview of current block queue where each character in the string represents a block set in the queue. `. = requested but not recieved`, `o = set received`, `m  = received set that matches the next blocks needed`
```
Comment (moneromooo-monero) [inline resources/developer-guides/daemon-rpc.md:1277]: s/recieved/received/

**Review by moneromooo-monero — COMMENTED** [review]
Just a few really minor nits, but it's good to go either way

**Review by moneromooo-monero — APPROVED** [review]

**Pull request merged**